### ISSUE TITLE
Generic class expressions in JS should issue an error

### ISSUE THREAD

**Issue by weswigham** [microsoft/TypeScript#31683]
**TypeScript Version:**  3.4.0-dev.201xxxxx

**Search Terms:**

**Code**

```ts
// @ts-check
class Foo {
	static Bar = class<T> {
		/** @type {number} */
		x;
	};
}
```

**Expected behavior:**
Error on `<T>`

**Actual behavior:**
No error.

`getJSSyntacticDiagnosticsForFile` in `program.ts` seems to be missing something.

**Comment by tanhauhau** [comment]
Can I have a try on this?

**Comment by RyanCavanaugh** [comment]
@tanhauhau sure!

**Comment by fatcerberus** [comment]
Out of curiosity what does TS do with the type parameter here?  Just ignore it?  Or does it actually contribute to type inference/checking?

**Comment by karthikkp** [comment]
@weswigham @RyanCavanaugh can you point to a mdn doc that talks about this?

**Comment by weswigham** [comment]
For clarity: we parse & check most TS constructs even in .js files, but we're _supposed_ to isssue a "X are not valid in a JavaScript file" error, where in this case, X is "Type parameters".

**Comment by karthikpriyadarshan-alation** [comment]
@weswigham So you mean when checking js files, generic class expressions should not be allowed and should throw an error. Where as in typescript this is fine?

**Comment by karthikkp** [comment]
@weswigham You mean, when the typescript compiler is used against a .js file which contains a generic class expression should throw an error?

**Comment by weswigham** [comment]
Report a diagnostic, similar to what we do when we find type parameters on a call expression. I referenced the function which needs adjustment in the OP.
